src/gradata/hooks/session_persist.py

```
@@ -0,0 +1,81 @@
+                nums = re.findall(r"\d+", line)
+                if nums:
+                    return int(nums[-1])
+    except Exception:
+        pass
+    return None
+
+
+def _get_modified_files() -> list[str]:
+    """Get files modified in current session via git diff."""
+    try:
+        result = subprocess.run(
```

Ruff (0.15.9)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)